packages/api-common/src/manifest-mixin.ts

```diff
@@ -19,10 +19,10 @@ export const emptyDataWarning = html`
   <div part="warning">No custom elements found in the JSON file.</div>
 `;
 
-export const ApiViewerMixin = <T extends Constructor<LitElement>>(
+export const ManifestMixin = <T extends Constructor<LitElement>>(
```

Comment:
Should we prefer this to be a controller?

If so we can also defer that change to another PR

Reply:
This mixin defines some properties so I'm not sure how to handle it with controller.

Reply:
can we move those to setters on the controller?

Reply:
or have the controller read host attrs for initial setup state?

Reply:
we could have the controller constructor receive an adapter object with methods that translate between controller state and host state, perhaps

Reply:
Feel free to create an issue if you think it's worth to get rid of the mixin. I have plans to change some components from Lit to vanilla in the future, so this code will need to be updated acordingly to remove `@property`.